timesketch/api/v1/resources/sigma.py

```diff
@@ -153,11 +217,371 @@ def post(self):
+        `/sigmarule/<string:rule_uuid>/` and returns a JSON represantion of the
+        rule.
+
+        Args:
+            rule_uuid: uuid of the rule
+
+        Returns:
+            JSON sigma rule representation
+            e.g.:  {"objects": [return_rule], "meta": {}}
+        """
+        try:
+            rule = SigmaRule.query.filter_by(rule_uuid=rule_uuid).first()
+
+        except Exception as e:  # pylint: disable=broad-except
+            error_msg = "Unable to get the Sigma rule {0!s}".format(e)
+            logger.error(
+                error_msg,
+                exc_info=True,
+            )
+            abort(
+                HTTP_STATUS_CODE_INTERNAL_SERVER_ERROR,
+                error_msg,
+            )
+
+        if not rule:
+            abort(HTTP_STATUS_CODE_NOT_FOUND, "No rule found with this ID.")
+        return_rule = []
+
+        assert isinstance(rule, SigmaRule)
+
+        return_rule.append(_enrich_sigma_rule_object(rule=rule))
+
+        return jsonify({"objects": return_rule, "meta": {}})
+
+    @login_required
+    def delete(self, rule_uuid):
+        """Deletes a Sigma rule from the database,.
+
+        Deletes a single Sigma rule selected by the `uuid` in
+        `/sigmarule/<string:rule_uuid>/` and returns a HTTP Code 200 if
+        the rule is deleted.
+
+        Args:
+            rule_uuid: uuid of the rule to be deleted
+
+        Returns:
+            HTTP_STATUS_CODE_NOT_FOUND if rule not found
+            HTTP_STATUS_CODE_OK if rule deleted
+        """
+
+        rule = SigmaRule.query.filter_by(rule_uuid=rule_uuid).first()
+
+        if not rule:
+            error_msg = "No rule found with rule_uuid.{0!s}".format(rule_uuid)
+            logger.debug(error_msg)  # only needed in debug cases
+            abort(
+                HTTP_STATUS_CODE_NOT_FOUND,
+                error_msg,
+            )
+
+        db_session.delete(rule)
+        db_session.commit()
+
+        return HTTP_STATUS_CODE_OK
+
+    @login_required
+    def post(self, rule_uuid=None):
+        """Adds a single Sigma rule to the database.
+
+        Adds a single Sigma rule to the database when `/sigmarule/` is called
+        with a POST request.
+
+        All attributes of the rule are taken by the `rule_yaml` value in the
+        POST request.
+
+        If no `rule_yaml` is found in the reuqest, the method will fail as this
+        is required to parse the rule.
+
+        If `rule_uuid` does not match the id provided in the yaml file the
+        request will fail.
+
+        Remark: To update a rule, use `PUT`instead.
+
+        Args:
+            rule_uuid: optional field that is not needed
+
+        Returns:
+            Sigma rule object and HTTP status 401 code indicating
+            whether operation was sucessful.
+            HTTP Error code 400 if no `rule_yaml` is provided
+            HTTP Error code 400 if rule_uuid does not match id in the yaml.
```

Comment:
You're also returning HTTP_STATUS_CODE_CONFLICT, so I'd encourage you to add a description here if you're also mentioning the other scenarios. Maybe also just something like:

```suggestion
            HTTP Error code 400 if no `rule_yaml` is provided or a parsing error occurs.
```